base/src/main/kotlin/io/goooler/demoapp/base/core/BaseFragment.kt

```diff
@@ -1,38 +1,32 @@
 package io.goooler.demoapp.base.core
 
-import android.view.KeyEvent
-import android.view.View
+import android.os.Bundle
+import androidx.activity.addCallback
 import androidx.fragment.app.Fragment
 
 abstract class BaseFragment : Fragment(), IFragment {
 
-  override fun onBackPressed(): Boolean {
-    activity?.onBackPressed()
-    return true
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    dispatchBackPress()
   }
 
   override fun finish() {
     activity?.finish()
   }
-
-  override fun onResume() {
-    super.onResume()
-    view?.dispatchBackPress()
-  }
 }
 
 sealed interface IFragment {
 
-  fun onBackPressed(): Boolean
+  fun onBackPressed(): Boolean = false
 
   fun finish()
 
-  fun View.dispatchBackPress() {
-    isFocusableInTouchMode = true
-    requestFocus()
-    setOnKeyListener { _, keyCode, event ->
-      val flag = keyCode == KeyEvent.KEYCODE_BACK && event.action == KeyEvent.ACTION_UP
-      if (flag) onBackPressed() else false
```

Comment:
Added `OnKeyListener` on Fragment's contentView, if contentView lost focus, this callback is useless.